### PR TITLE
chore(test-data): create `CartDiscountTarget` models and implement in presets

### PR DESCRIPTION
**Pull request by jaikamat**
## Summary

This PR creates only the relevant `CartDiscountTarget` models required to maintain our test data models.

References for review:
- [Sample data](https://github.dev/commercetools/test-data-research/blob/6d2d1fa5babc37b0abbccf474c6fd6490f029db6/src/entity-drafts/cart-discounts.ts#L12)
- [HTTP API documentation](https://docs.commercetools.com/api/projects/cartDiscounts#cartdiscounttarget)
- The `CartDiscountTarget` schema is viewable on Impex

## Description

This was a little more involved than I anticipated, mostly due to size of boilerplate and the requirement to create a whole new set of models in the same style as `/commons`

Some takeaways:
1. The HTTP API does not have a Draft representation for `CartDiscountTarget`
2. The GraphQL API _does_ have a Draft representation for `CartDiscountTarget`, known as `CartDiscountTargetInput`
3. Each GraphQL model within `CartDiscountTargetInput` has its own unique draft representation

What this means is that I had to create draft models colocated with each non-draft model, and we only use them for their GraphQL transformations.

Consider the HTTP representation for `MultiBuyLineItemsTarget` (property values/types omitted):
```
{
  type: 'multiBuyLineItems',
  predicate,
  triggerQuantity,
  discountedQuantity,
  maxOccurrence,
  selectionMode,
}
```

The GraphQL representation for this used in a draft context is is:
```
{
  multiBuyLineItems: {
    type: 'multiBuyLineItems',
    predicate,
    triggerQuantity,
    discountedQuantity,
    maxOccurrence,
    selectionMode,
    __typename: 'MultiBuyLineItemsTargetInput',
  },
  __typename: 'CartDiscountTargetInput',
}
```

### CODE REVIEW

**Comment by jaikamat** [comment]
Recently restructured this PR given @valoriecarli's move to colocate models (great thinking!)

```
cart-discount/
├─ src/
│  ├─ cart-discount/
│  ├─ cart-discount-line-items-target/
│  ├─ cart-discount-multi-buy-line-items-target/
│  ├─ cart-discount-shipping-cost-target/
```

**Review by tylermorrisford — APPROVED** [review]
🏆

File: models/cart-discount/src/cart-discount/cart-discount-draft/presets/sample-data-fashion/percent-off-womens-wear.ts [hunk 1]
```diff
@@ -9,8 +10,11 @@ const percentOffWomensWear = (): TCartDiscountDraftBuilder =>
     .empty()
     .value(CartDiscountValueRelativeDraft.random().permyriad(1500))
     .cartPredicate('1 = 1')
-    // TODO: create `target` model
-    .target(undefined)
+    .target(
+      CartDiscountLineItemsTargetDraft.random().predicate(
+        `categories.key contains "bottoms-women" and price.discount.id is not defined`
```
Comment (stephsprinkle) [inline models/cart-discount/src/cart-discount/cart-discount-draft/presets/sample-data-fashion/percent-off-womens-wear.ts:15]: Is it worth merging the category preset PR in then updating the key here with the preset?

File: models/cart-discount/src/cart-discount-shipping-cost-target/cart-discount-shipping-cost-target-draft/generator.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Generator } from '@commercetools-test-data/core';
+import { TCartDiscountShippingCostTargetDraft } from '../types';
+
+// https://docs.commercetools.com/api/projects/cartDiscounts#cartdiscountshippingcosttarget
+
+const generator = Generator<TCartDiscountShippingCostTargetDraft>({
+  fields: {
+    type: 'shipping',
```
[the remark this answers is not in the document]
Comment (jaikamat) [reply models/cart-discount/src/cart-discount-shipping-cost-target/cart-discount-shipping-cost-target-draft/generator.ts:8]: Sigh, right? 🤣

**Comment by jaikamat** [comment]
@stephsprinkle thanks for the feedback on the loose ends wrt keys. I'll rebase on top and get this straightened out

File: models/cart-discount/README.md [hunk 3]
```diff
@@ -12,6 +12,8 @@ $ yarn add -D @commercetools-test-data/cart-discount
 
 # Usage
 
+## `CartDiscount`
+
 ```ts
 import type {
   TCartDiscount,
```
[the remark this answers is not in the document]
Comment (jaikamat) [reply models/cart-discount/README.md]: Thanks! c73d2ee48a694082cc0483f7bc5b4ab20bd72ce4

**Comment by jaikamat** [comment]
@stephsprinkle see fbbb3011dbee91ee802352f0becb2ed32e239c85

**Review by stephsprinkle — APPROVED** [review]

**Pull request merged**